rmgpy/tools/merge_models.py

```diff
@@ -160,4 +161,17 @@ def combine_models(models):
                   '#{0:d}.'.format(i+1, Nrxn - Nrxn0, len(model.reactions)))
     print('The merged model has {0:d} species and {1:d} reactions'
           ''.format(len(final_model.species), len(final_model.reactions)))
+
+    # ensure no species with same name and index
+    label_index_dict = {}
+    for s in final_model.species:
+        if s.label not in label_index_dict.keys():
```

Comment:
This can just be `if s.label not in label_index_dict`. No need to call `keys()`.

Reply:
good to know! Just made the code simpler!